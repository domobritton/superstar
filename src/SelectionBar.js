import React from 'react';
import styled from 'styled-components';

import AppBar from '@material-ui/core/AppBar';
import Tabs from '@material-ui/core/Tabs';
import Tab from '@material-ui/core/Tab';

export const SelectionBar = ({ handleChange, value }) => {
  return (
    <AppBar position="static" color="default">
      <Tabs
        value={value}
        onChange={handleChange}
        indicatorColor="primary"
        textColor="primary"
        variant="fullWidth"
      >
        <Select label="View My Stars" />
        <Select label="Search By User" />
        <Select label="Search By Repo" />
      </Tabs>
    </AppBar>
  );
};

const Select = styled(Tab)`
  span {
    font-size: 17px;
    @media (max-width: 600px) {
      font-size: 12px;
    }
  }
`;
