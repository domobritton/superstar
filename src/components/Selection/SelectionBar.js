import React from 'react';
import PropTypes from 'prop-types';

import AppBar from '@material-ui/core/AppBar';
import Tabs from '@material-ui/core/Tabs';
import { Select } from './SelectionBarStyle';

export const SelectionBar = ({ handleChange, value }) => (
  <AppBar position="static" color="default">
    <Tabs
      value={value}
      onChange={handleChange}
      indicatorColor="primary"
      textColor="primary"
      variant="fullWidth"
    >
      <Select label="View My Stars" />
      <Select label="Search By User" />
      <Select label="Search By Repo" />
    </Tabs>
  </AppBar>
);

SelectionBar.propTypes = {
  handleChange: PropTypes.func,
  value: PropTypes.number,
};
