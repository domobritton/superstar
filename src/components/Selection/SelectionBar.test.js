import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { SelectionBar } from './SelectionBar';

afterEach(cleanup);

test('<SelectionBar />', () => {
  const { container } = render(<SelectionBar />);
  expect(container.firstChild).toMatchSnapshot();
});
