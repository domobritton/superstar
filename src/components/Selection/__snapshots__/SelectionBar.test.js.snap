// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectionBar /> 1`] = `
<header
  class="MuiPaper-root-10 MuiPaper-elevation4-16 MuiAppBar-root-1 MuiAppBar-positionStatic-5 MuiAppBar-colorDefault-7"
>
  <div
    class="MuiTabs-root-37"
  >
    <div
      class="MuiTabs-flexContainer-38"
    >
      <div
        class="MuiTabs-scroller-40 MuiTabs-fixed-41"
        role="tablist"
        style="margin-bottom: 0px;"
      >
        <div
          class="MuiTabs-flexContainer-38"
        >
          <button
            aria-selected="false"
            class="MuiButtonBase-root-58 MuiTab-root-46 MuiTab-textColorPrimary-49 MuiTab-fullWidth-53 sc-bdVaJa bCGswX"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper-54"
            >
              <span
                class="MuiTab-labelContainer-55"
              >
                <span
                  class="MuiTab-label-56"
                >
                  View My Stars
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root-61"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root-58 MuiTab-root-46 MuiTab-textColorPrimary-49 MuiTab-fullWidth-53 sc-bdVaJa bCGswX"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper-54"
            >
              <span
                class="MuiTab-labelContainer-55"
              >
                <span
                  class="MuiTab-label-56"
                >
                  Search By User
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root-61"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root-58 MuiTab-root-46 MuiTab-textColorPrimary-49 MuiTab-fullWidth-53 sc-bdVaJa bCGswX"
            role="tab"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiTab-wrapper-54"
            >
              <span
                class="MuiTab-labelContainer-55"
              >
                <span
                  class="MuiTab-label-56"
                >
                  Search By Repo
                </span>
              </span>
            </span>
            <span
              class="MuiTouchRipple-root-61"
            />
          </button>
        </div>
        <span
          class="MuiPrivateTabIndicator-root-68 MuiPrivateTabIndicator-colorPrimary-69 MuiTabs-indicator-45"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
  </div>
</header>
`;
